src/main/PersistentState.cpp

```diff
@@ -36,47 +38,94 @@ PersistentState::dropAll(Database& db)
     st.execute(true);
 }
 
-string
+std::string
 PersistentState::getStoreStateName(PersistentState::Entry n)
 {
     if (n < 0 || n >= kLastEntry)
     {
         throw out_of_range("unknown entry");
     }
-    return mapping[n];
+    auto res = mapping[n];
```

Comment:
this is pretty ugly and also introduces a bug.

First the bug:
you're not restoring SCP state after an upgrade.

The badness:
having `getStoreStateName` rely on side effects is super counter intuitive.

I suggest instead that you add an optional parameter `int subscript` (default 0) that, when non 0, gets appended to the name of the key (otherwise don't append anything).

Reply:
Fixed.